clients/bigquery/tableid.go

```diff
@@ -0,0 +1,41 @@
+package bigquery
+
+import (
+	"fmt"
+
+	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/sql"
+)
+
+type TableIdentifier struct {
+	projectID string
+	dataset   string
+	table     string
+}
+
+func NewTableIdentifier(projectID, dataset, table string) TableIdentifier {
+	return TableIdentifier{projectID: projectID, dataset: dataset, table: table}
+}
+
+func (ti TableIdentifier) ProjectID() string {
+	return ti.projectID
+}
+
+func (ti TableIdentifier) Dataset() string {
+	return ti.dataset
+}
+
+func (ti TableIdentifier) Table() string {
+	return ti.table
+}
+
+func (ti TableIdentifier) FullyQualifiedName(escape, uppercaseEscNames bool) string {
```

Comment:
Compare to https://github.com/artie-labs/transfer/pull/450/files#diff-b939896b6f41e55cd45d2840b3436928e963e5eb3d50abf58a3edbe267f37fb6L54